rfcs/0000-changes-lifecyle.md

```diff
@@ -0,0 +1,205 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [FuelLabs/sway-rfcs#0000](https://github.com/FuelLabs/sway-rfcs/pull/001)
+- Sway Issue: [FueLabs/sway#0000](https://github.com/FuelLabs/sway/issues/001)
+
+# Summary
+
+[summary]: #summary
+
+This RFC intends to propose a scheme to track changes in the compiler. 
+
+# Motivation
+
+[motivation]: #motivation
+
+The compiler needs a guide on how to introduce changes to improve developer experience every time a compiler is updated.
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+## PR changes
+
+Every PR will have a section named "Release Notes", which will contains a user friendly explanation of what the PR contains.
+
+The compiler release notes **WILL** be the amalgamation of these section "as is".
+
+## Feature flags
+
+Any complex change that needed to be gated will need the following steps.
+
+1 - A specific github issue labeled with `feature-*` and `track-feature`. This issue should be the umbrella for everything related with this feature;
+2 - A preliminary PR enabling the feature flag should be created and linked in to the umbrella issue; This PR will enable the feature flag `--experimental <comma-separated-list>` on all tools.
+3 - As many PRs will be created and merged an normal;
+4 - When the feature is ready, a closing PR will be created and wait until the feature flag is enabled by default.
+5 - On a later date, the feature flag can be removed making the feature the default behavior of the compiler. 
+
+# Enabling features on `sway`
+
+Features can be enabled inside the `Forc.toml` file:
+
+```toml
+experimental = ["...", "..."]
+```
+
+or using the CLI
+
+```
+> forc ... --experimental some_feature,another_feature
+```
+
+These flags also need to be enabled programmatically by any compiler driver, like tests.
+
+Unlike `Rust` we will not support features inside sway code like the example below, because some features will span across multiple tools. That would demand `forc` to parse, or ask the `sway` compiler if a feature is enabled or not. 
+
+```sway
+#![enable(some_experimental_feature)]
+```
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC intends to propose a scheme to track changes in the compiler. These changes will be categorized into two ways:
+
+1 - Non-breaking changes;
+2 - Breaking changes.
+
+Breaking changes will be any changes that, after updating the compiler version, causes:
+
+1 - Code without deprecated warning to be flagged as error;
+2 - ABI json properties to be removed or have their type changed;
+3 - SDK (rust, typescript and others) to break;
```

Comment:
This includes it implicitly but we should explicitly mention noncompatible changes to the ABI (not the spec, the actual binary interface of the contracts) and VM incompatibility.